### PR TITLE
[Test] Add draft long-haul test cases

### PR DESCRIPTION
**Pull request by vinli-cn**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Enable daily run long-haul test case execution using Github action.

Fault test and Update test are added, but they would be updated further
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: test/long-haul/fault-test.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+print_logs_title "Reset AKS environment and start sample workload"
+reset_csi_driver
+start_sample_workload
+sleep $SleepInSecs
+verify_csi_driver
+verify_sample_workload workloadPodName workloadNodeName
+
+
+print_logs_title "Delete workload pod and verify new workload pod "
+kubectl delete po $workloadPodName
```
Comment (joe-atzinger) [inline test/long-haul/fault-test.sh:38]: Should this be "pod"?

File: test/long-haul/fault-test.sh [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+print_logs_title "Verify workload pod on worker node"
+verify_sample_workload workloadPodNameNew workloadNodeNameNew
+if [[ "$workloadPodName" != "$workloadPodNameNew" || "$workloadNodeName" != "$workloadNodeNameNew" ]] ; then
+    print_logs_error "expected workload pod $workloadPodName on $workloadNodeName, actual new workload pod $workloadPodNameNew on $workloadNodeNameNew"
+    signal_err
+fi
+
+
+print_logs_title "Delete the workload pod on the worker node and verify its state"
+kubectl delete po $workloadPodName > /dev/null 2>&1 &
```
Comment (joe-atzinger) [inline test/long-haul/fault-test.sh:92]: "pod"?

File: test/long-haul/utils.sh [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+reset_csi_driver () {
+    kubectl replace -f $Repo/deploy/csi-azurelustre-node.yaml
+
+    kubectl get nodes --no-headers | grep "$PoolName" | awk '{print $1}' | 
+    {
+        while read n; 
+        do
+            kubectl label nodes $n node4faulttest-
```
Comment (joe-atzinger) [inline test/long-haul/utils.sh:40]: Do you mean to have the trailing dash?

**Comment by vinli-cn** [comment]
@andyzhangx pls complete this PR

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**